grails-forge-core/src/main/java/org/grails/forge/feature/github/workflows/plain/templates/plainGithubWorkflow.rocker.raw

```diff
@@ -12,23 +12,31 @@ on: [push, pull_request]
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      checks: write
+      contents: read
     steps:
 @javaSetup.template(jdkVersion)
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@@v4
+        # Uncomment the build-scan parameters if you want to publish a Gradle build scan
+        # in order to see all the build logs, a complete task timeline, test outputs,
+        # and the resolved dependencies of your build.
+        # with:
+        #   build-scan-publish: true
+        #   build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
+        #   build-scan-terms-of-use-agree: "yes"
       - name: Run Tests
         if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch'
         id: tests
-        uses: gradle/gradle-build-action@@v2
-        with:
-          arguments: check
+        run: ./gradlew check
       - name: Run Build
         if: github.event_name == 'push'
         id: build
-        uses: gradle/gradle-build-action@@v2
-        with:
-          arguments: build
+        run: ./gradlew build
       - name: Publish Test Report
-        if: steps.build.outcome == 'failure' || steps.tests.outcome == 'failure'
+        if: failure()
         uses: scacap/action-surefire-report@@v1
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
From the `action-surefire-report` README: (https://github.com/ScaCap/action-surefire-report?tab=readme-ov-file#github_token)
> Optional. Usually in form of github_token: ${{ secrets.GITHUB_TOKEN }}. The default is ${{ github.token }}.

As the default is `github.token`, isn't this param redundant and can be removed?

Reply:
Turns out it can be removed. I tested this out in one of my repositories:
https://github.com/tylervz/adding-build-info/actions/runs/10361951888/job/28683138274